package/benchmarks/zdt/_zdt.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+from typing import Any
+
+import optproblems.zdt
+import optuna
+import optunahub
+
+
+class Problem(optunahub.benchmarks.BaseProblem):
+    """Wrapper class for the WFG test suite of optproblems."""
+
+    def __init__(self, function_id: int, **kwargs: Any) -> None:
+        """Initialize the problem.
+        Args:
+            function_id: Function ID of the WFG problem in [1, 6].
+            kwargs: Arbitrary keyword arguments, please refer to the optproblems documentation for more details.
+
+        Please refer to the optproblems documentation for the details of the available properties.
+        https://www.simonwessing.de/optproblems/doc/zdt.html
+        """
+        assert 1 <= function_id <= 6, "function_id must be in [1, 6]"
+        self._problem = {
+            1: optproblems.zdt.ZDT1,
+            2: optproblems.zdt.ZDT2,
+            3: optproblems.zdt.ZDT3,
+            4: optproblems.zdt.ZDT4,
+            5: optproblems.zdt.ZDT5,
+            6: optproblems.zdt.ZDT6,
+        }[function_id](**kwargs)
+
+        if function_id in [1, 2, 3]:
+            self._search_space = {
+                f"x{i}": optuna.distributions.FloatDistribution(0.0, 1.0) for i in range(30)
+            }
+        elif function_id in [4, 6]:
+            self._search_space = {
+                f"x{i}": optuna.distributions.FloatDistribution(0.0, 1.0) for i in range(10)
+            }
+        else:
+            # x0 \in {0, 1}^{30}
+            self._search_space = {"x0": optuna.distributions.IntDistribution(0, 1 << 29)}
+            # x1, ..., x11 \in {0, 1}^{5}
+            for i in range(1, 11):
+                self._search_space[f"x{i}"] = optuna.distributions.IntDistribution(0, 1 << 4)
```

Comment:
```suggestion
            self._search_space = {}
            for i, binary_length in enumerate([30] + [5]*10):
                self._search_space.update(
                    {
                        f"x{i}_{b}": optuna.distributions.CategoricalDistribution([True, False])
                        for b in range(binary_length)
                    }
                )
```

Reply:
IMO, this search space should be lists of booleans rather than integers because the distance cannot be treated appropriately in optimizers.